apps/web/pages/api/teams/api/create.ts

```diff
@@ -0,0 +1,118 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import type Stripe from "stripe";
+import { z } from "zod";
+
+import stripe from "@calcom/features/ee/payments/server/stripe";
+import { HttpError } from "@calcom/lib/http-error";
+import { defaultHandler, defaultResponder } from "@calcom/lib/server";
+import prisma from "@calcom/prisma";
+import { MembershipRole } from "@calcom/prisma/enums";
+import { _MembershipModel as Membership, _TeamModel as Team } from "@calcom/prisma/zod";
+
+const querySchema = z.object({
+  session_id: z.string().min(1),
+});
+
+const checkoutSessionMetadataSchema = z.object({
+  pendingPaymentTeamId: z.string().transform(Number),
+  ownerId: z.string().transform(Number),
+});
+
+type CheckoutSessionMetadata = z.infer<typeof checkoutSessionMetadataSchema>;
+
+export const schemaTeamReadPublic = Team.omit({});
+export const schemaMembershipPublic = Membership.merge(z.object({ team: Team }).partial());
+
+async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const checkoutSession = await getCheckoutSession(req);
+  validateCheckoutSession(checkoutSession);
+  const checkoutSessionSubscription = getCheckoutSessionSubscription(checkoutSession);
+  const checkoutSessionMetadata = getCheckoutSessionMetadata(checkoutSession);
+
+  const activatedTeam = await prisma.team.update({
+    where: { id: checkoutSessionMetadata.pendingPaymentTeamId },
+    data: {
+      pendingPayment: false,
+    },
+  });
+
+  const finalizedTeam = await prisma.team.update({
+    where: { id: checkoutSessionMetadata.pendingPaymentTeamId },
+    data: {
+      members: {
+        create: {
+          userId: checkoutSessionMetadata.ownerId as number,
+          role: MembershipRole.OWNER,
+          accepted: true,
+        },
+      },
+      metadata: {
+        ...(typeof activatedTeam.metadata === "object" ? activatedTeam.metadata : {}),
+        paymentId: checkoutSession.id,
+        subscriptionId: checkoutSessionSubscription.id || null,
+        subscriptionItemId: checkoutSessionSubscription.items.data[0].id || null,
+      },
+    },
+    include: { members: true },
+  });
```

Comment:
I don't think a new team with payment pending will have any significant metadata. Also the requetedSlug is deprecated anyways 

```suggestion
  const finalizedTeam = await prisma.team.update({
    where: { id: checkoutSessionMetadata.pendingPaymentTeamId },
    data: {
      pendingPayment: false,
      members: {
        create: {
          userId: checkoutSessionMetadata.ownerId as number,
          role: MembershipRole.OWNER,
          accepted: true,
        },
      },
      metadata: {
        paymentId: checkoutSession.id,
        subscriptionId: checkoutSessionSubscription.id || null,
        subscriptionItemId: checkoutSessionSubscription.items.data[0].id || null,
      },
    },
    include: { members: true },
  });
```

Reply:
Agreed. Done in [Ignore pending team metadata](https://github.com/calcom/cal.com/pull/13211/commits/d7a66ccd4832b84af56b4abf6e7e686b55e89357)